src/EditorFeatures/Core/Implementation/Formatting/FormatCommandHandler.FormatDocument.cs

```diff
@@ -93,6 +97,24 @@ public bool ExecuteCommand(FormatDocumentCommandArgs args, CommandExecutionConte
                 {
                     var docOptions = document.GetOptionsAsync(c.CancellationToken).WaitAndGetResult(c.CancellationToken);
 
+                    // AB Test
+                    // If the feature is OFF and the feature options have not yet been set to their assigned flight, do so now
+                    if (!docOptions.GetOption(CodeCleanupOptions.PerformAdditionalCodeCleanupDuringFormatting)
+                        && !docOptions.GetOption(CodeCleanupABTestOptions.ParticipatedInExperiment))
+                    {
+                        var experimentationService = document.Project.Solution.Workspace.Services.GetService<IExperimentationService>();
+
+                        if (experimentationService != null)
+                        {
+                            if (experimentationService.IsExperimentEnabled("CleanupOn"))
+                            {
+                                docOptions.WithChangedOption(new OptionKey(CodeCleanupOptions.PerformAdditionalCodeCleanupDuringFormatting, LanguageNames.CSharp), true);
```

Comment:
will this work? I think you need to do: docOptions = docOptions.WithChangedOption(...